spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/MultiRabbitBootstrapConfiguration.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.annotation;
+
+import org.springframework.amqp.rabbit.config.RabbitListenerConfigUtils;
+import org.springframework.amqp.rabbit.listener.RabbitListenerEndpointRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.annotation.ImportBeanDefinitionRegistrar;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotationMetadata;
+import org.springframework.lang.Nullable;
+
+/**
+ * An {@link ImportBeanDefinitionRegistrar} class that registers
+ * a {@link MultiRabbitListenerAnnotationBeanPostProcessor} bean, if MultiRabbit
+ * is enabled.
+ *
+ * @author Wander Costa
+ *
+ * @since 1.4
+ *
+ * @see RabbitListenerAnnotationBeanPostProcessor
+ * @see MultiRabbitListenerAnnotationBeanPostProcessor
+ * @see RabbitListenerEndpointRegistry
+ * @see EnableRabbit
+ */
+public class MultiRabbitBootstrapConfiguration implements ImportBeanDefinitionRegistrar, EnvironmentAware {
+
+	private Environment environment;
+
+	@Override
+	public void registerBeanDefinitions(@Nullable AnnotationMetadata importingClassMetadata,
+			BeanDefinitionRegistry registry) {
+
+		if (isMultiRabbitEnabled() && !registry.containsBeanDefinition(
```

Comment:
Hi @garyrussell, to give you some background on this class, I initially implemented a slightly different version of [your suggestion](https://github.com/spring-projects/spring-amqp/pull/1111#discussion_r563866329), centralizing the condition for both AutoConfig (at boot) and for the MultiRabbitBPP.